src/code42cli/cmds/devices.py

```diff
@@ -435,23 +465,56 @@ def bulk(state):
 def bulk_deactivate(state, csv_rows, change_device_name, purge_date, format):
     """Deactivate all devices from the provided CSV containing a 'guid' column."""
     sdk = state.sdk
-    csv_rows[0]["deactivated"] = False
-    formatter = OutputFormatter(format, {key: key for key in csv_rows[0].keys()})
     for row in csv_rows:
         row["change_device_name"] = change_device_name
         row["purge_date"] = purge_date
 
+    def _deactivate(row):
+        _deactivate_device(
+            sdk, row["guid"], row["change_device_name"], row["purge_date"]
+        )
+
+    _bulk_handle_device_activation_update(
+        sdk, format, csv_rows, _deactivate, "deactivate", "Deactivating"
+    )
+
+
+def _bulk_handle_device_activation_update(
```

Comment:
This feels like one of those places that extracting out some minor duplicate logic into a function just makes it harder to read/understand what's happening. I don't see any problem with just copying the bulk_deactivate command and tweaking the little bit that's necessary to make it work for reactivate, rather than try to share code between the two commands.

Reply:
Done